sw/device/lib/runtime/irq.c

```diff
@@ -0,0 +1,150 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/runtime/irq.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdnoreturn.h>
+
+#include "sw/device/lib/uart.h"
+
+#define IRQ_ABI __attribute__((aligned(4), interrupt))
+
+size_t irq_type_scratch_area = 0;
+IRQ_ABI void irq_internal_dispatch(void) {
+  irq_handler_t handler =
+      irq_vtable[irq_type_scratch_area / sizeof(irq_handler_t)];
+  uart_send_str("Trampolining to interrupt handler at 0x");
```

Comment:
should these uart_send_str be changed to the new LOG usage? Otherwise if we use this in our DV environment, we will probably experience quite a bit of uart serializing slow down.